lib/src/facts/collection.cc

```diff
@@ -319,20 +329,33 @@ namespace facter { namespace facts {
         return false;
     }
 
+    void collection::resolve(shared_ptr<resolver> const& res) {
+        remove(res);
+
+        // Check if the resolver has been blocked
+        if (try_block(res)) {
+            return;
+        }
+
+        // Check if the resolver should be cached
+        auto resolver_ttl = _ttls.find(res->name());
+        if (resolver_ttl != _ttls.end()) {
+           cache::use_cache(*this, res, (*resolver_ttl).second);
```

Comment:
This happens before resolving facts. So I don't understand how this will result in caching the correct values if the cache is deemed invalid.